Riot/AppDelegate.m

```diff
@@ -1421,41 +1577,88 @@ - (void)handleLocalNotificationsForAccount:(MXKAccount*)account
                     // to include a percent symbol (%) in the message, use two percent symbols (%%).
                     notificationBody = [notificationBody stringByReplacingOccurrencesOfString:@"%" withString:@"%%"];
 
-                    UILocalNotification *eventNotification = [[UILocalNotification alloc] init];
-                    eventNotification.alertBody = notificationBody;
-                    eventNotification.userInfo = @{
-                                                   @"type": @"full",
-                                                   @"room_id": event.roomId,
-                                                   @"event_id": event.eventId,
-                                                   @"user_id": account.mxCredentials.userId
-                                                   };
+                    if (@available(iOS 10, *))
```

Comment:
Note:
We could factorize for the 2 APIs like this:
```
NSDictionary *notificationUserInfo = @{
        @"type": @"full",
        @"room_id": event.roomId,
        @"event_id": event.eventId,
        @"user_id": account.mxCredentials.userId
};

BOOL isNotificationContentShown = !event.isEncrypted || RiotSettings.shared.showDecryptedContentInNotifications;

NSString *categoryIdentifier;

if ((event.eventType == MXEventTypeRoomMessage || event.eventType == MXEventTypeRoomEncrypted) && isNotificationContentShown)
{
    categoryIdentifier = @"QUICK_REPLY";
}

NSString *soundName;

// Set sound name based on the value provided in action of MXPushRule
for (MXPushRuleAction *action in rule.actions)
{
    if (action.actionType == MXPushRuleActionTypeSetTweak && [action.parameters[@"set_tweak"] isEqualToString:@"sound"])
    {
        NSString *soundName = action.parameters[@"value"];
        if ([soundName isEqualToString:@"default"])
            soundName = @"message.mp3";     
    }
}
```

Reply:
https://github.com/vector-im/riot-ios/pull/2207/commits/589829d0765f3a392bce5ed3d962a0489f441765